### PR TITLE
Fixes curator whip disarming even after getting parried

### PR DESCRIPTION
**Pull request by grungussuss**
## About The Pull Request
what the title says
### Test:

https://github.com/tgstation/tgstation/assets/96586172/e68323e0-2f44-42db-9752-4c4b5d5c4a3c
## Changelog
:cl:
fix: curator whip will no longer disarm when parried
/:cl:

### CODE REVIEW

File: code/game/objects/items/melee/misc.dm [hunk 1]
```diff
@@ -330,13 +330,14 @@
 	attack_verb_simple = list("flog", "whip", "lash", "discipline")
 	hitsound = 'sound/weapons/whip.ogg'
 
-/obj/item/melee/curator_whip/afterattack(target, mob/user, proximity_flag)
-	. = ..()
+/obj/item/melee/curator_whip/attack(target, mob/user, proximity_flag)
```
Comment (MrMelbert) [inline code/game/objects/items/melee/misc.dm:333]:
```suggestion
/obj/item/melee/curator_whip/attack(mob/living/target, mob/living/user, params)
```

File: code/_onclick/item_attack.dm [hunk 2]
```diff
@@ -234,7 +234,9 @@
 		user.client.give_award(/datum/award/achievement/misc/selfouch, user)
 
 	user.do_attack_animation(target_mob)
-	target_mob.attacked_by(src, user)
+
```
Comment (MrMelbert) [inline code/_onclick/item_attack.dm:237]:
```suggestion
```

**Review by MrMelbert — APPROVED** [review]
Probably fine, probably

**Comment by necromanceranne** [comment]
What if we had this whole mechanic that allowed you to stop attacks from hitting you.

What if we had a 15 force melee weapon that not only ignored that mechanic, but also removed anything in your hand. And you can do this endlessly.

~~Effectively keeping three officers from being even able to make an attack against me as I remove all their equipment from their hands and slash them to pieces is the single stupidest thing that the curator can do and I think people don't abuse this as much as they could purely out of respect for their fellow player.~~

**Comment by DaCoolBoss** [comment]
This has been due for a long time.

**Comment by zoomachina** [comment]
Could this be related to an issue where embeds don't care about the fact the projectile has been blocked?

https://github.com/tgstation/tgstation/assets/97964454/d71e2800-d9a6-4a9e-b3f4-eb7b506a46e4

**Comment by Sadboysuss** [comment]
> Could this be related to an issue where embeds don't care about the fact the projectile has been blocked?
> 
>  blockbad.mp4

it's something with projectile block code and is a different matter

**Comment by Sakamoto4ka** [comment]
A unique weapon that can be 1 on shift and used as a counter block will now be blocked

**Comment by MrMelbert** [comment]
> Could this be related to an issue where embeds don't care about the fact the projectile has been blocked?

No, this is just largely an issue with the fact that a lot of our code is written on the assumption that X happens so when X doesn't end up happening... we take an L

File: code/_onclick/item_attack.dm [hunk 3]
```diff
@@ -234,7 +234,8 @@
 		user.client.give_award(/datum/award/achievement/misc/selfouch, user)
 
 	user.do_attack_animation(target_mob)
-	target_mob.attacked_by(src, user)
+	if(!target_mob.attacked_by(src, user))
```
Comment (MrMelbert) [inline code/_onclick/item_attack.dm:237]: You should make this proc return `TRUE` if force <= 0 rather than `FALSE`, it'll fix CI, and makes sense I guess

Comment (Sadboysuss) [reply code/_onclick/item_attack.dm:237]: done

File: code/game/objects/items/melee/misc.dm [hunk 4]
```diff
@@ -330,13 +330,14 @@
...
+/obj/item/melee/curator_whip/attack(mob/living/target, mob/living/user, params)
+	.= ..()
+	if(.)
+		return
+	if(ishuman(target))
 		var/mob/living/carbon/human/human_target = target
 		human_target.drop_all_held_items()
 		human_target.visible_message(span_danger("[user] disarms [human_target]!"), span_userdanger("[user] disarmed you!"))
-
 /obj/item/melee/roastingstick
```
Comment (Jacquerel) [inline code/game/objects/items/melee/misc.dm:341]:
```suggestion

/obj/item/melee/roastingstick
```

File: code/game/objects/items/melee/misc.dm [hunk 5]
```diff
@@ -330,13 +330,14 @@
 	attack_verb_simple = list("flog", "whip", "lash", "discipline")
 	hitsound = 'sound/weapons/whip.ogg'
 
-/obj/item/melee/curator_whip/afterattack(target, mob/user, proximity_flag)
-	. = ..()
-	if(ishuman(target) && proximity_flag)
+/obj/item/melee/curator_whip/attack(mob/living/target, mob/living/user, params)
+	.= ..()
```
Comment (Jacquerel) [inline code/game/objects/items/melee/misc.dm:334]:
```suggestion
	. = ..()
```

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request merged**